### DISCUSSION TITLE
Auto-importing methods in the outer init file

### DISCUSSION THREAD

**Discussion by Hilly12** [synthesized-io/fairlens#101] [General]
At the moment the workflow for most methods in fairlens is

```python
import fairlens as fl
 
fl.plot.distr_plot()
```

Perhaps for packages `sensitive` and `plot` its worth including them in the top level `__init__` file so we can call them without specifying the package name.

```python
import fairlens as fl

fl.distr_plot() 
```

I think we can keep the `metrics` package as a separate import though.

```python
from fairlens import metrics as fm
 
fm.EarthMoversDistance(x, y)
```

Thoughts?

**Comment by bogdansurdu** [comment]
I think it would be fine to do so for `sensitive`, but not too sure about `plot`. The reason for this is that in a normal workflow we use very few methods from `sensitive`, whereas there are multiple methods in `plot` that we might want to use to generate visualizations.

So for example we could have:

```python
import fairlens as fl

fl.detect_names_df(df, deep_search=True)
```

**Comment by Hilly12** [comment]
The idea is to completely remove the `.package` which shouldn't make a huge difference since we're already importing all of these methods anyway. The reason for not doing this for `metrics` was because it's a significantly larger package than the rest. I think the methods in `plot` are quite simple and modular so shouldn't conflict with anything else or complicate things if that's what you meant.

**Comment by rob-tay** [comment]
tbh, I would be more in favour of changing the names of some of the functions rather than the import structure.  I don't really see `fl.plot.distr_plot` as being an issue...it's only an extra four characters and keeps the structure of the packages explicit.

**Reply by Hilly12** [reply]
Yeah ngl would be much better if it was something on the grounds of `fl.plot.distributions()`. Tbf having second thoughts, I think this refactor would require big changes to api reference so maybe let's leave it at this for now.
